geode-core/src/main/java/org/apache/geode/internal/cache/AbstractUpdateOperation.java

```diff
@@ -179,22 +186,32 @@ public static boolean doPutOrCreate(LocalRegion rgn, EntryEventImpl ev, long las
             br.getPartitionedRegion().getPrStats().startApplyReplication();
           }
           try {
-            if (rgn.basicUpdate(ev, false/* ifNew */, true/* ifOld */, lastMod,
-                overwriteDestroyed)) {
+            boolean secondUpdateSuccess;
```

Comment:
maybe "firstUpdateSuccess" is a better name, since there're 2 updates here.

Reply:
The first update attempt takes place on line 155. Would it be worth extracting that method call to a boolean similar to second- and thirdUpdateSuccess, for consistency?

Reply:
We have renamed the booleans to first-, second- and thirdBasicUpdateSuccess for each of the three calls to `LocalRegion.basicUpdate` in the method, for clarity.